### PR TITLE
Spitter Weakrefs

### PR DESCRIPTION
**Pull request by Misttyy**
## About The Pull Request

Changes Spitter Hardrefs to Weakrefs

## Why It's Good For The Game

Cleaner code, less HardDels, less HardRefs. Helps me learn refactoring on a basic scale

## Changelog

:cl: Mistfrag
refactor: Spitters no longer use HardRefs for their abilities, now use WeakRefs.
:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/castes/Spitter.dm [hunk 1]
```diff
@@ -74,11 +74,10 @@
...
 			return
 
-	for (var/atom/dotA in dot_cooldown_atoms)
-		if (dotA == A)
-			return
+	if(WEAKREF (A) in dot_cooldown_atoms)
+		return
 
-	dot_cooldown_atoms += A
+	dot_cooldown_atoms += WEAKREF (A)
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/castes/Spitter.dm:80]:
```suggestion
	dot_cooldown_atoms += WEAKREF(A)
```

**Review by stanalbatross — COMMENTED** [review]
`dot_cooldown_atoms -= A`

should this not be weakref?

**Comment by Misttyy** [comment]
> `dot_cooldown_atoms -= A`
> 
> 
> 
> should this not be weakref?

Will fix this tomorrow when I'm home.

**Comment by Misttyy** [comment]
@stanalbatross Added

**Comment by Misttyy** [comment]
a

File: code/modules/mob/living/carbon/xenomorph/castes/Spitter.dm [hunk 2]
```diff
@@ -74,11 +74,10 @@
 		if (H.stat == DEAD)
 			return
 
-	for (var/atom/dotA in dot_cooldown_atoms)
-		if (dotA == A)
-			return
+		if(WEAKREF(A) in dot_cooldown_atoms)
+		return
```
Comment (morrowwolf) [inline code/modules/mob/living/carbon/xenomorph/castes/Spitter.dm:78]: At a quick glance I assume this return is meant to be indented so as it be in the if()

**Review by morrowwolf — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Pull request merged**